.github/workflows/validate-MacOS-binaries.yml

```diff
@@ -0,0 +1,116 @@
+name: Validate MacOS Binaries
+
+on:
+  pull_request:
+    paths:
+      - .github/workflows/validate-MacOS-binaries.yml
+
+
+jobs:
+  generate-conda-matrix:
+    uses: pytorch/test-infra/.github/workflows/generate_binary_build_matrix.yml@main
+    with:
+      package-type: conda
+      os: macos
+      channel: all
+      test-infra-ref: weiwangmeta/add-macos-x86_64-support
+
+  validate-MacOS-conda-packages:
+    needs: generate-conda-matrix 
+    runs-on: ${{ matrix.validation_runner }} 
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJSON(needs.generate-conda-matrix.outputs.matrix) }}
+    env: 
+      MAC_ARCH:  ${{ matrix.validation_runner }}
+      PYTHON_VERSION: ${{ matrix.python_version }} 
+      CON_CHANNEL: ${{ matrix.channel }}
+    steps:
+      - name: Checkout PyTorch/Builder directory
+        uses: actions/checkout@v3
+      - name: Install Fresh Miniconda in Runner Temp Dir
+        run: |
+          set -ex
+          rm -rf ${{ runner.temp }}/*
+          if [ ${MAC_ARCH} == 'macos-m1-12' ]; then
+            cd ${{ runner.temp }}
+            curl -O https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-arm64.sh
+            sh Miniconda3-latest-MacOSX-arm64.sh -b -f  -p ${{ runner.temp }}
+          elif [ ${MAC_ARCH} == 'macos-12-xl' ]; then
+            cd ${{ runner.temp }}
+            curl -O https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh
+            sh Miniconda3-latest-MacOSX-x86_64.sh -b -f  -p ${{ runner.temp }}
+          fi 
+      - name: Conda Install pytorch and smoke test
+        env:
+          ENV_NAME: conda-env-${{ github.run_id }}
+        run: |
+          set -ex
+          conda create -yp ${ENV_NAME} python=${PYTHON_VERSION} numpy
```

Comment:
Could you please get installation instructions from matrix ? Then your test become :
```
 - name: Conda Install pytorch and smoke test
        env:
          GPU_ARCH_VER: ${{ matrix.gpu_arch_version }}
          GPU_ARCH_TYPE: ${{ matrix.gpu_arch_type }}
          INSTALLATION: ${{ matrix.installation }}
          ENV_NAME: conda-env-${{ github.run_id }}
        run: |
          set -ex
          conda create -yp ${ENV_NAME} python=${{ matrix.python_version }} numpy
          conda run -p ${ENV_NAME} $INSTALLATION
          conda run -p ${ENV_NAME} python3  ./test/smoke_test/smoke_test.py
```

Reply:
Thanks for the suggestion, let me try.

Reply:
This is done, and tests pass.